tests/sockets_cores_threads_test.go

```diff
@@ -0,0 +1,484 @@
+	} `yaml:"spec"`
+}
+
+// Return true if there is CPU field in requests or limits, and return these fields
+func parseYAMLConfig(podYAML string) (bool, int, int) {
+	isCPUPresent := false
+	resourcesRequests := 0
+	resourcesLimits := 0
+	var config Config
+	yaml.Unmarshal([]byte(podYAML), &config)
+
+	// Checking that fields Requests and Limits exist
+	if len(config.Spec.Containers[0].Resources.Requests) != 0 && len(config.Spec.Containers[0].Resources.Limits) != 0 {
+		_, doesElementExist := config.Spec.Containers[0].Resources.Requests["cpu"]
+		if doesElementExist {
+			isCPUPresent = true
+			resourcesRequests, _ = strconv.Atoi(config.Spec.Containers[0].Resources.Requests["cpu"])
+		}
+
+		_, doesElementExist = config.Spec.Containers[0].Resources.Requests["cpu"]
+		if doesElementExist {
+			isCPUPresent = true
+			resourcesLimits, _ = strconv.Atoi(config.Spec.Containers[0].Resources.Limits["cpu"])
+		}
+	}
+
+	return isCPUPresent, resourcesRequests, resourcesLimits
+}
+
+// Checking if the cluster can run at least one VM
+func isEnoughResources(virtClient kubecli.KubevirtClient, cpuNeeded int, memNeeded int64) (bool, int, int) {
+	availableVMs, cpu_limit, mem_limit := tframework.GetAvailableResources(virtClient, int64(cpuNeeded), memNeeded)
+	if availableVMs == 0 {
+		fmt.Println("no resources avaible", "cpuNeeded=", cpuNeeded, "memNeeded=", memNeeded)
+		return false, cpu_limit, mem_limit
+
+	} else {
+		fmt.Println("possible to run VM", "cpuNeeded=", cpuNeeded, "memNeeded=", memNeeded)
+		return true, cpu_limit, mem_limit
+
+	}
+}
+
+func getYAMLFilename(sockets, cores, threads int, address_common string) string {
+	// 0 mens parameter set to 0, 1 means parameter set to non-zero
+	var file_name [2][2][2]string
+	file_name[0][0][0] = "vm-template-fedora-no-sockets-cores-and-threads.yaml"
+	file_name[1][0][0] = "vm-template-fedora-only-sockets.yaml"
+	file_name[0][1][0] = "vm-template-fedora-only-cores.yaml"
+	file_name[0][0][1] = "vm-template-fedora-only-threads.yaml"
+	file_name[0][1][1] = "vm-template-fedora-only-cores-and-threads.yaml"
+	file_name[1][0][1] = "vm-template-fedora-only-sockets-and-threads.yaml"
+	file_name[1][1][0] = "vm-template-fedora-only-sockets-and-cores.yaml"
+	file_name[1][1][1] = "vm-template-fedora-sockets-cores-and-threads.yaml"
+
+	// Because go doesn't have ternary operators
+	s := 0
+	c := 0
+	t := 0
+	if sockets > 0 {
+		s = 1
+	}
+	if cores > 0 {
+		c = 1
+	}
+	if threads > 0 {
+		t = 1
+	}
+
+	return address_common + file_name[s][c][t]
+}
+
+func getArguments(vm_name string, sockets int, cores int, threads int) []string {
+	arguments := []string{"NAME=" + vm_name}
+
+	if sockets != 0 {
+		arguments = append(arguments, "CPU_SOCKETS="+strconv.Itoa(sockets))
+	}
+	if cores != 0 {
+		arguments = append(arguments, "CPU_CORES="+strconv.Itoa(cores))
+	}
+	if threads != 0 {
+		arguments = append(arguments, "CPU_THREADS="+strconv.Itoa(threads))
+	}
+
+	return arguments
+}
+
+var _ = Describe("numaSupport", func() {
```

Comment:
NUMA? This has nothing to do with numa.. guest topology might be better name here. Also shouldn't you annotate this with the test plan ID somehow so the results are automatically connected with it?

Reply:
>NUMA
Just old variable, I'll rename it. I've already rename everything about NUMA, but in this place forget to do it

>Also shouldn't you annotate this with the test plan ID somehow so the results are automatically connected with it?

I'm going to do this work it in another PR for all my test plans together. Now I just want to merge this code ASAP  (before it outdated, and before it requires rebase/difficult merges, etc)

Reply:
Please fix

Reply:
It was already fixed